### DISCUSSION TITLE
Using Revisions and appending additional (meta)data?

### DISCUSSION THREAD

**Discussion by ryanbuening** [ravendb/ravendb#16769] [Q&A] [answered]
Is there a way to append additional data to a revision? For example, if we want to record "modified by" user information on each revision.

**Comment by ayende** [comment, marked as answer]
You cannot modify a _revision_, but you can update the metadata on document write, which will be saved (and in the revision as well):

https://ravendb.net/docs/article-page/5.4/csharp/client-api/session/how-to/subscribe-to-events#onbeforestore

**Reply by ryanbuening** [reply]
Oh cool. Now I just need to find a spot in .NET core to subscribe to that at a place where the user claims are accessible, but will also still get ran every time the app runs.

**Comment by georgiosd** [comment]
There is both a global and an injectable http context, so you should be
able to do it pretty much anywhere

On Wed, 28 Jun 2023 at 18:59, Ryan Buening ***@***.***> wrote:

> Oh cool. Now I just need to find a spot in .NET core to subscribe to that
> at a place where the user claims are accessible, but will also still get
> ran every time the app runs.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ravendb/ravendb/discussions/16769#discussioncomment-6306607>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMYG5VN3HVOBQQSRVOEOQ3XNRIGDANCNFSM6AAAAAAZXJCQZM>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
-- 
Georgios Diamantopoulos
Founder
Zero to MVP
https://zerotomvp.dev
